### PR TITLE
configure: check for C11 atomics support

### PR DESCRIPTION
**Pull request by sitsofe**
So I know you've merged something already to address this but I was fiddling with about with this in the morning to check if it worked. I figure I'll post it for posterity even if it doesn't get merged (I'd have to rebase it anyway).

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -550,6 +550,26 @@ else
...
+int main(void)
+{
+  _Atomic unsigned v;
+  atomic_load(&v);
+  return 0;
+}
+EOF
+if ! compile_prog "" "" "C11 atomics"; then
+  echo
+  echo "From fio 3.21 compiler support for C11 atomics is required."
```
Comment (axboe) [inline configure:566]: I'd remove the version info here, it isn't really relevant. It'll just end up being a historic thing that's then removed later, and it doesn't even really add anything right now. They obviously have this version (or newer) if they run into this :-)

**Comment by axboe** [comment]
Honestly, I just pushed out what I had, since I didn't hear back. I'm fine with this one too, and it's probably better since it doesn't rely on version checking.

File: configure (hunk 1 above)
Comment (sitsofe) [reply configure]: Fair point - done.

**Comment by sitsofe** [comment]
OK updated - hopefully this looks better...

**Comment by axboe** [comment]
LGTM!

**Comment by axboe** [comment]
LGTM!

**Pull request merged**